### DISCUSSION TITLE
[Question]: Add loading indicator for file uploads in LibreChat to prevent system freeze perception.

### DISCUSSION THREAD

**Discussion by angpao** [danny-avila/LibreChat#4479] [Q&A]
### What is your question?

Loading

### More Details

“To enhance the display of a loading indicator during file uploads in LibreChat, you can improve the user interface to notify users that the file is being uploaded, preventing the system from appearing frozen. This can be done by adding a loading element or animation in the file upload section.”

### Screenshots
<img width="339" alt="Screenshot 2567-10-21 at 11 19 11" src="https://github.com/user-attachments/assets/f3ca5376-969d-4237-b0e4-3d9b23de193f">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
I've already improved the loading animation, but it's indicative since we're simulating the ETA. We may need to revise this again with a more accurate loading spinner

**Reply by pascalmariany** [reply]
Hi Berry! We are facing issues. Users get frustated and asking why their files aren't recognized by the AI Agent, even if they upload it twice or they wait a bit. Seems the indexing takes long. Could you have a look into this?

**Comment by angpao** [comment]
Awesome , When can I test that?

**Comment by danny-avila** [comment]
The upload can be cancelled now, hovering over the file and clicking the "X" icon.

**Reply by berry-13** [reply]
he probably just wanted a better loading spinner progress. We could estimate the upload progress using the file size and upload speed, but tracking index progress is harder. It would mean changing the RAG API, LibreChat’s backend, and setting up a webhook for real-time updates @angpao

**Reply by angpao** [reply]
Yes, I want to show spinner progress until finish.
